src/controllers/nodes/nodes_controller.go

```diff
@@ -227,17 +227,26 @@ func (controller *NodesController) handleOutdatedCache(nodeCache *Cache) error {
 	}
 
 	for _, cachedNodeName := range nodeCache.Keys() {
+		var cachedNodeInCluster = false
 		for _, clusterNode := range nodeLst.Items {
 			if clusterNode.Name == cachedNodeName {
 				cachedNodeInfo, err := nodeCache.Get(cachedNodeName)
 				if err != nil {
 					log.Error(err, "failed to get node", "node", cachedNodeName)
 					return err
 				}
+				cachedNodeInCluster = true
+				// Check if node was seen less than an hour ago, otherwise do not remove from cache
 				controller.removeNodeFromCache(nodeCache, cachedNodeInfo, cachedNodeName)
 				break
```

Comment:
could you return out of the function here and always remove the node from the cache if the loop is fully traversed?
would remove the need for the variable

Reply:
guess not, need to go through all the nodes 😄

Reply:
yes, otherwise first node which is found ends the function